plugins/elanfp/fu-elanfp-usb-device.c

```diff
@@ -0,0 +1,564 @@
+/*
+ * Copyright (C) 2021
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <glib/gstdio.h>
+
+#include "fu-elanfp-file-control.h"
+#include "fu-elanfp-usb-device.h"
+
+#define ELAN_EP_CMD_OUT	     (0x01 | 0x00)
+#define ELAN_EP_CMD_IN	     (0x02 | 0x80)
+#define ELAN_EP_MOC_CMD_IN   (0x04 | 0x80)
+#define ELAN_EP_IMG_IN	     (0x03 | 0x80)
+#define ELANFP_USB_INTERFACE 0
+
+#define CTRL_SEND_TIMEOUT_MS		 3000
+#define BULK_SEND_TIMEOUT_MS		 1000
+#define BULK_RECV_TIMEOUT_MS		 1000
+#define ELANFP_FLASH_TRANSFER_BLOCK_SIZE 1024
+
+#define PROTOCOL_VERSION_2 0x02
+#define PROTOCOL_VERSION_4 0x04
+
+#define FIRMWARE_UPDATE_OFFER_SKIP   0x00
+#define FIRMWARE_UPDATE_OFFER_ACCEPT 0x01
+#define FIRMWARE_UPDATE_OFFER_REJECT 0x02
+
+#define FIRMWARE_UPDATE_FLAG_FIRST_BLOCK 0x80
+#define FIRMWARE_UPDATE_FLAG_LAST_BLOCK	 0x40
+
+#define STA_REJECT_OLD_FIRMWARE	      0x00
+#define STA_REJECT_SWAP_PENDING	      0x02
+#define STA_REJECT_WRONG_BANK	      0x04
+#define STA_REJECT_SIGN_RULE	      0xE0
+#define STA_REJECT_VER_RELEASE_DEBUG  0xE1
+#define STA_REJECT_DEBUG_SAME_VERSION 0xE2
+
+#define FIRMWARE_UPDATE_SUCCESS		   0x00
+#define FIRMWARE_UPDATE_ERROR_WRITE	   0x02
+#define FIRMWARE_UPDATE_ERROR_VERIFY	   0x04
+#define FIRMWARE_UPDATE_ERROR_SIGNATURE	   0x06
+#define FIRMWARE_UPDATE_ERROR_INVALID_ADDR 0x09
+#define FIRMWARE_UPDATE_ERROR_NO_OFFER	   0x0A
+#define FIRMWARE_UPDATE_ERROR_INVALID	   0x0B
+
+#define REPORT_ID_FW_VERSION_FEATURE 0x20
+#define REPORT_ID_OFFER_COMMAND	     0x25
+#define REPORT_ID_OFFER_RESPONSE     0x25
+#define REPORT_ID_PAYLOAD_COMMAND    0x20
+#define REPORT_ID_PAYLOAD_RESPONSE   0x22
+
+#define REQTYPE_GET_VERSION 0xC1
+#define REQTYPE_COMMAND	    0x41
+
+struct _FuElanfpUsbDevice {
+	FuUsbDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuElanfpUsbDevice, fu_elanfp_usb_device, FU_TYPE_USB_DEVICE)
+
+// Communication
+gboolean
+iapSendCommand(GUsbDevice *usb_device,
```

Comment:
please use _ rather than snake case

Reply:
Fixed in commit 5af9c23594ac261d85fbf53fca02de08a5ef0150.